src/librustdoc/core.rs

```diff
@@ -75,6 +76,8 @@ crate struct DocContext<'tcx> {
     /// See `collect_intra_doc_links::traits_implemented_by` for more details.
     /// `map<module, set<trait>>`
     crate module_trait_cache: RefCell<FxHashMap<DefId, FxHashSet<DefId>>>,
+    /// Fake empty cache used when cache is required as parameter.
+    crate cache: Cache,
```

Comment:
Hmm, this isn't exactly right - the cache is filled at some point in the process, it's just not filled *immediately*. I think the comment belongs on `DocContext::new`, not here.

Reply:
No, the `cache` in `DocContext` is never filled.

Reply:
I am confused. If the cache is never filled, why is it needed?

Reply:
Let me find you \*the\* place haha.

Reply:
Here: https://github.com/rust-lang/rust/pull/80987/files#diff-df9f90aae0b7769e1eb6ea6f1d73c1c3f649e1ac48a20e169662a8930eb427beR1402

Reply:
That link anchor doesn't work for me.

Reply:
Ah sorry, it changed in the meantime. So to prevent the link to break again, take a look at:

```rust
impl Type {
    fn inner_def_id(&self, cache: Option<&Cache>) -> Option<DefId> {
```

in `clean/types.rs`. The whole mess started because we needed the `Cache` type to get the primitive types' `DefId`. And starting from there, I went up and fixed all compiler errors and you can see the diff now. :rofl: :sob: